src/main/java/com/es/phoneshop/service/ProductService.java

```diff
@@ -1,15 +1,23 @@
 package com.es.phoneshop.service;
 
 import com.es.phoneshop.model.product.Product;
+import com.es.phoneshop.model.product.sorting.SortField;
+import com.es.phoneshop.model.product.sorting.SortOrder;
 
 import java.util.List;
 
 public interface ProductService {
 
     Product getProduct(Long id);
 
+    Product getProductByCode(String code);
+
     List<Product> findProducts();
```

Comment:
is this method used anywhere? it should be when there's no search/sort field, but in servlet I don't see it's invocation